v10.5 Changelog (#7704)
mattermost/docs: source/about/mattermost-v10-changelog.md

- See the [changelog in progress](https://bit.ly/2nK3cVf) for details about the upcoming release.
```

(release-v10.5-extended-support-release)=
## Release v10.5 - [Extended Support Release](https://docs.mattermost.com/about/release-policy.html#release-types)

**Release Day: Februrary 14, 2025**

### Compatibility
 - Updated minimum Safari version to 17.4+ and minimum Firefox version to 119+.

### Upgrade
 - v10.5 introduces Property System Architecture schema migration. See the [Important Upgrade Notes](https://docs.mattermost.com/upgrade/important-upgrade-notes.html) for details.

>  - Improved plugins’ link tooltip component’s internal architecture without any user interface changes.

Author: @M-ZubairAhmed Is there a better description for the breaking change introduced by https://github.com/mattermost/mattermost/pull/29727.

Reviewer: The internal workings of the PluginLinkComponent in the webapp have been improved. With this plugins that register link tooltips using registerLinkTooltipComponent will see changes in how tooltip components are managed. Link tooltips are now unmounted from the DOM by default, significantly improving performance. They are only mounted when a link is hovered over or focused. As a result, plugins need to update their components to properly handle mounting and unmounting scenarios.

>  - The internal workings of the ``PluginLinkComponent`` in the webapp have been improved. Plugins that register link tooltips using ``registerLinkTooltipComponent`` will see changes in how tooltip components are managed. Link tooltips are now unmounted from the DOM by default, significantly improving performance. They are only mounted when a link is hovered over or focused. As a result, plugins need to update their components to properly handle mounting and unmounting scenarios.

Reviewer 2: Should this be communicated as a breaking change? There may be no in-use third party plugins affected, but if there are, this is a breaking change for them.

>  - The Mattermost server has stopped supporting manual plugin deployment. Plugins are deployed manually when an administrator or some deployment automation copies the contents of a plugin bundle into the server's working directory. If a manual or automated deployment workflow is still required, administrators can instead prepackage the plugin bundles. See more details in [this forum post](https://forum.mattermost.com/t/deprecation-notice-manual-plugin-deployment/21192).

Reviewer 2: ```suggestion
 - The Mattermost server has stopped supporting manual plugin deployment. Plugins were deployed manually when an administrator or some deployment automation copies the contents of a plugin bundle into the server's working directory. If a manual or automated deployment workflow is still required, administrators can instead prepackage the plugin bundles. See more details in [this forum post](https://forum.mattermost.com/t/deprecation-notice-manual-plugin-deployment/21192).
```

```{Important}
If you upgrade from a release earlier than v10.3, please read the other [Important Upgrade Notes](https://docs.mattermost.com/upgrade/important-upgrade-notes.html).
```

### Improvements

#### User Interface (UI)
 - Pre-packaged Calls plugin [v1.5.0](https://github.com/mattermost/mattermost-plugin-calls/releases/tag/v1.5.0).
 - Pre-packaged MS Teams plugin [v2.1.0](https://github.com/mattermost/mattermost-plugin-msteams/releases/tag/v2.1.0).
 - Pre-packaged Channel Export plugin [v1.2.1](https://github.com/mattermost/mattermost-plugin-channel-export/releases/tag/v1.2.1).
 - Added the ability to modify post attachments during edit.
 - The channel bookmarks bar is now hidden when there are no bookmarks in the channel. Bookmarks can now be added from the channel menu.
 - Removed the video from the onboarding checklist.
 - Improved accessibility of many form labels throughout the app.

#### Administration
 - Added the migrations, store layer and service for the Property System Architecture.
 - Added the Custom Profile Attribute fields store, app and API endpoints.
 - Added the **System Console** user interface for managing Custom Profile Attributes (User Properties).

>  - Added a compliance export overhaul. See announcement: <TBD>.

Author: @cpoile Is there a link to the announcement?

 - Introduced V2 of the Support Packet, containing improvement diagnosis information for high-availability deployments.
 - Added a new ``Fallback`` field to ``PluginSettingsSection`` that controls whether the settings defined under the section should still render as fallback when the plugin is disabled.
 - Updated the library used for tooltips throughout the app to fix a memory leak.
 - Reduced the volume of unnecessary debug logs generated during scheduled post job execution.
 - Removed ``form-data`` from @mattermost/client.

### Bug Fixes
 - Fixed archived filter behavior in System Console > User Management > Channels to restore the ability to exclude archived channels.

>  - Fixed an issue where DMs/GMs with a `DeleteAt` flag in the database might cause issues with a couple APIs.

Reviewer 2: ```suggestion
 - Fixed an issue where DMs/GMs with a `DeleteAt` non-zero value in the database might cause issues with several APIs.
```

 - v10.5 introduces Property System Architecture schema migration. See the [Important Upgrade Notes](https://docs.mattermost.com/upgrade/important-upgrade-notes.html) for details.
 - The Compliance Export system has been overhauled. See the [Important Upgrade Notes](https://docs.mattermost.com/upgrade/important-upgrade-notes.html) for details.
 - The Mattermost server has stopped supporting manual plugin deployment. Plugins were deployed manually when an administrator or some deployment automation copies the contents of a plugin bundle into the server's working directory. If a manual or automated deployment workflow is still required, administrators can instead prepackage the plugin bundles. See more details in [this forum post](https://forum.mattermost.com/t/deprecation-notice-manual-plugin-deployment/21192).
 - Mattermost has stopped official Mattermost server builds for the Microsoft Windows operating system. Administrators should migrate existing Mattermost server installations to use the official Linux builds. See more details in [this forum post](https://forum.mattermost.com/t/deprecation-notice-server-builds-for-microsoft-windows/21498).

### Breaking Changes

>  - The internal workings of the ``PluginLinkComponent`` in the webapp have been improved. Plugins that register link tooltips using ``registerLinkTooltipComponent`` will see changes in how tooltip components are managed. Link tooltips are now unmounted from the DOM by default, significantly improving performance. They are only mounted when a link is hovered over or focused. As a result, plugins need to update their components to properly handle mounting and unmounting scenarios.

Reviewer 2: ```suggestion
 - The internal workings of the ``PluginLinkComponent`` in the webapp have been improved. Plugins that register link tooltips using ``registerLinkTooltipComponent`` will see changes in how tooltip components are managed. Link tooltips are now unmounted from the DOM by default, significantly improving performance. They are only mounted when a link is hovered over or focused. As a result, plugins may need to update their components to properly handle mounting and unmounting scenarios.
```

Not all plugins are affected.

 - v10.5 introduces Property System Architecture schema migration. See the [Important Upgrade Notes](https://docs.mattermost.com/upgrade/important-upgrade-notes.html) for details.
 - The Compliance Export system has been overhauled. See the [Important Upgrade Notes](https://docs.mattermost.com/upgrade/important-upgrade-notes.html) for details.
 - The Mattermost server has stopped supporting manual plugin deployment. Plugins were deployed manually when an administrator or some deployment automation copies the contents of a plugin bundle into the server's working directory. If a manual or automated deployment workflow is still required, administrators can instead prepackage the plugin bundles. See more details in [this forum post](https://forum.mattermost.com/t/deprecation-notice-manual-plugin-deployment/21192).
 - Mattermost has stopped official Mattermost server builds for the Microsoft Windows operating system. Administrators should migrate existing Mattermost server installations to use the official Linux builds. See more details in [this forum post](https://forum.mattermost.com/t/deprecation-notice-server-builds-for-microsoft-windows/21498).

### Breaking Changes

>  - The internal workings of the ``PluginLinkComponent`` in the webapp have been improved. Plugins that register link tooltips using ``registerLinkTooltipComponent`` will see changes in how tooltip components are managed. Link tooltips are now unmounted from the DOM by default, significantly improving performance. They are only mounted when a link is hovered over or focused. As a result, plugins may need to update their components to properly handle mounting and unmounting scenarios.

Reviewer 3: ```suggestion
 - The internal workings of the ``PluginLinkComponent`` in the webapp have been changed to unmounted link tooltips from the DOM by default, significantly improving performance. Plugins that register link tooltips using ``registerLinkTooltipComponent`` will see changes in how tooltip components are managed. They are only mounted when a link is hovered over or focused. As a result, plugins may need to update their components to properly handle mounting and unmounting scenarios.
```

@M-ZubairAhmed can we be more specific on what plugin developers need to update here? We can possibly link the example of what was done in the GH plugin.

Reviewer: thanks @marianunez added more information as expected above

```{Important}
If you upgrade from a release earlier than v10.3, please read the other [Important Upgrade Notes](https://docs.mattermost.com/upgrade/important-upgrade-notes.html).
```

### Improvements

#### User Interface (UI)
 - Pre-packaged Boards plugin [v9.1.0](https://github.com/mattermost/mattermost-plugin-boards/releases/tag/v9.1.0).
 - Pre-packaged Calls plugin [v1.5.1](https://github.com/mattermost/mattermost-plugin-calls/releases/tag/v1.5.1).
 - Pre-packaged MS Teams plugin [v2.1.0](https://github.com/mattermost/mattermost-plugin-msteams/releases/tag/v2.1.0).
 - Pre-packaged Channel Export plugin [v1.2.1](https://github.com/mattermost/mattermost-plugin-channel-export/releases/tag/v1.2.1).
 - Pre-packaged Jira plugin [v4.2.0](https://github.com/mattermost/mattermost-plugin-jira/releases/tag/v4.2.0).
 - Added the ability to modify post attachments during edit.
 - The channel bookmarks bar is now hidden when there are no bookmarks in the channel. Bookmarks can now be added from the channel menu.
 - Removed the video from the onboarding checklist.

>  - Improved accessibility of many form labels throughout the app.

Reviewer 3: ```suggestion
 - Improved accessibility throughout the webapp by fixing several issues around keyboard navigation and screen reader focused on modals, RHS and core chat functionality. 
```
